eclair-node/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -224,9 +224,13 @@ trait Service extends ExtraDirectives with Logging {
                           }
                         } ~
                         path("sendtonode") {
-                          formFields(amountMsatFormParam, paymentHashFormParam, nodeIdFormParam, "maxAttempts".as[Int].?, "feeThresholdSat".as[Satoshi].?, "maxFeePct".as[Double].?, "externalId".?) {
-                            (amountMsat, paymentHash, nodeId, maxAttempts_opt, feeThresholdSat_opt, maxFeePct_opt, externalId_opt) =>
+                          formFields(amountMsatFormParam, paymentHashFormParam.?, nodeIdFormParam, "maxAttempts".as[Int].?, "feeThresholdSat".as[Satoshi].?, "maxFeePct".as[Double].?, "externalId".?, "keysend".?) {
```

Comment:
nit: let's group optional parameters together (and keysend should be typed):

```suggestion
                          formFields(amountMsatFormParam, nodeIdFormParam, paymentHashFormParam.?, "keysend".as[Boolean].?, "maxAttempts".as[Int].?, "feeThresholdSat".as[Satoshi].?, "maxFeePct".as[Double].?, "externalId".?) {
```

I'm wondering if it makes sense to allow users to provide the preimage themselves (in which case "keysend" would be a `ByteVector32` and we wouldn't generate it ourselves in `sendWithPreimage`). It's probably fine to start with just a `Boolean` and that can evolve if users ask for the ability to choose the preimage themselves.

Reply:
Yes, I don't think it's needed right now. Nevertheless, I've modified `sendWithPreimage` to take an optional `paymentPreimage`. So if we want to give the user the opportunity to use their own preimage, we would just need to change `Boolean` to `ByteVector32`.

Reply:
Done